### PR TITLE
Add the ability to hide the chat from firestore

### PR DESCRIPTION
**Pull request by sunny-viktoryia**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1169

### CODE REVIEW

File: src/hooks/useSettings.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import { Settings } from "types/Firestore";
+import { SparkleSelector } from "types/SparkleSelector";
+
+import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+export const settingsSelector: SparkleSelector<Settings | undefined> = (
+  state
+) => state.firestore.data.settings;
```
Comment (goran-peoski-work) [inline src/hooks/useSettings.ts:15]: Move to `selectors.ts` as an ordered/hashed pair ?

Example:
```ts
export const ownedVenuesDataSelector: SparkleSelector<
  Record<string, AnyVenue> | undefined
> = (state) => state.firestore.data.ownedVenues;

export const ownedVenuesSelector: SparkleSelector<
  WithId<AnyVenue>[] | undefined
> = (state) => state.firestore.ordered.ownedVenues;
```

File: src/hooks/useSettings.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    { collection: "settings", doc: "customizations", storeAs: "settings" },
+  ]);
+
+  const settings = useSelector(settingsSelector);
+
+  return useMemo(
+    () => ({
+      showChat: settings?.showChat ?? DEFAULT_SHOW_CHAT,
+    }),
+    [settings]
```
Comment (goran-peoski-work) [inline src/hooks/useSettings.ts]: Few notes on this. Nothing mandatory.

1. Construction can be done in order:

```ts
()=>({
  ...DEFAULT_VALUES_FIRST,
  ...valuesFromDbThatExistWillOverwrite
})
```

2. The above, however, doesn't guard much with DB types being incorrect like getting a string when array or boolean is expected, but I guess for now a more generic solution might be OOS.
For the specific property `settings?.showChat` can be checked with `=== true` or `=== false` before applying 
the default, or just converted with `!!`.

3. Will this PR be followed up with another that adds setting for reactions as well?

File: src/types/Firestore.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ export interface Experience {
   tables: Record<string, Record<string, Table>>;
 }
 
+export interface Settings {
+  showChat?: boolean;
+}
```
Comment (goran-peoski-work) [inline src/types/Firestore.ts:24]: One more for https://github.com/sparkletown/internal-sparkle-issues/issues/1170 ?

2 issues for the price of 1 PR

**Review by goran-peoski-work — APPROVED** [review]
Few comments on code. Mostly LGTM. Could be use to resolve 2 issues in one PR though

File: src/hooks/useSettings.ts (hunk 1 above)
Comment (goran-peoski-work) [reply src/hooks/useSettings.ts:15]: Hm... looking at this, might not be feasible to have an ordered version I guess

File: src/hooks/useSettings.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      setAreLoaded(false);
+    } else {
+      setAreLoaded(true);
+    }
+    // disabling because we should be dependent only on settings here
+  }, [settings]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return useMemo(
+    () => ({
+      areLoaded: isTruthy(areLoaded),
```
Comment (mike-lvov) [inline src/hooks/useSettings.ts:41]:
```suggestion
      isSettingsLoaded: isLoaded(settings),
```

File: src/hooks/useSettings.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { useEffect, useMemo, useState } from "react";
+
+import { DEFAULT_SHOW_CHAT } from "settings";
+
+import { Settings } from "types/Firestore";
+import { SparkleSelector } from "types/SparkleSelector";
+
+import { isTruthy } from "utils/types";
+
+import { useFirestoreConnect } from "hooks/useFirestoreConnect";
```
Comment (mike-lvov) [inline src/hooks/useSettings.ts:10]:
```suggestion
import { useFirestoreConnect, isLoaded } from "hooks/useFirestoreConnect";
```

**Review by mike-lvov — COMMENTED** [review]
haha

File: src/hooks/useSettings.ts (hunk 4 above)
Comment (sunny-viktoryia) [reply src/hooks/useSettings.ts]: haha, fixed 🙈 
the commit is here: https://github.com/sparkletown/sparkle/pull/2261/commits/e37f952af6f7c168cafd88fd5a8710c09d0bcfd6

Thank youuuuuu!! ❤️

File: src/hooks/useSettings.ts (hunk 5 above)
Comment (sunny-viktoryia) [reply src/hooks/useSettings.ts]: the commit is here: https://github.com/sparkletown/sparkle/pull/2261/commits/e37f952af6f7c168cafd88fd5a8710c09d0bcfd6

Thank youuuuuu!! ❤️

File: src/hooks/useSettings.ts (hunk 1 above)
Comment (mike-lvov) [reply src/hooks/useSettings.ts:13]: hashed version is enough. Move to selectors

Comment (sunny-viktoryia) [reply src/hooks/useSettings.ts:13]: why do we need ordered here? 😬

Comment (goran-peoski-work) [reply src/hooks/useSettings.ts:13]: Not saying we do, ( if you meant https://github.com/sparkletown/sparkle/pull/2261#discussion_r702970543 ). I just copied both as to keep the naming with the `Data` segment as unordered

File: src/types/Firestore.ts (hunk 3 above)
Comment (mike-lvov) [reply src/types/Firestore.ts:24]: I'd mark it as OOS

Comment (mike-lvov) [inline src/types/Firestore.ts:24]: move this from types/Firestore into separate type file

**Review by mike-lvov — APPROVED** [review]
LGTM + some neat stuff

File: src/hooks/useSettings.ts (hunk 1 above)
Comment (sunny-viktoryia) [reply src/hooks/useSettings.ts]: moved here: https://github.com/sparkletown/sparkle/pull/2261/commits/481c1df6d605841a96963ce3f6e73c9f01961154

File: src/types/Firestore.ts (hunk 3 above)
Comment (sunny-viktoryia) [reply src/types/Firestore.ts]: moved here: https://github.com/sparkletown/sparkle/pull/2261/commits/d42a8ec5d419219b9d48075684b1d1ceac94025b

**Pull request merged**